### PR TITLE
BACK-906: added new chains to existing Swapsicle integration

### PR DESCRIPTION
**Pull request by aburkut**
The PR is based on the changes from @rorrieswap 's PR - https://github.com/paraswap/paraswap-dex-lib/pull/238

### Summary of the changes:

- [x] add e2e  `Swapsicle` tests for `Polygon` chain https://github.com/paraswap/paraswap-dex-lib/pull/293#issuecomment-1416099992
- [x] add e2e `Swapsicle` tests for `Avalanche` chain https://github.com/paraswap/paraswap-dex-lib/pull/293#issuecomment-1416100931
- [x] add e2e `Swapsicle` tests for `BSC` chain https://github.com/paraswap/paraswap-dex-lib/pull/293#issuecomment-1416102130
- [x] add e2e `Swapsicle` tests for `Fantom` chain https://github.com/paraswap/paraswap-dex-lib/pull/293#issuecomment-1416122257
- [x] add e2e `Swapsicle` tests for `Mainnet` chain https://github.com/paraswap/paraswap-dex-lib/pull/293#issuecomment-1416152310
- [x] add e2e `Swapsicle` tests for `Optimisim` chain https://github.com/paraswap/paraswap-dex-lib/pull/293#issuecomment-1418693811
- [x] add e2e `Swapsicle` tests for `Arbitrum` chain https://github.com/paraswap/paraswap-dex-lib/pull/293#issuecomment-1418923785

### CODE REVIEW

**Comment by aburkut** [comment]
Polygon `Swapcicle` test results:

paraswap-dex-lib – uniswap-v2-e2e-polygon test ts 2023-02-03 16-12-54

**Comment by aburkut** [comment]
Avalanche `Swapcicle` test results:

paraswap-dex-lib – uniswap-v2-e2e-avalanche test ts 2023-02-03 18-44-55

**Comment by aburkut** [comment]
BSC chain `Swapcicle` test results:

paraswap-dex-lib – uniswap-v2-e2e-bsc test ts 2023-02-03 19-17-27

**Comment by aburkut** [comment]
Fatom chain `Swapcicle` test results:

paraswap-dex-lib – uniswap-v2-e2e-fantom test ts 2023-02-03 19-41-22

**Comment by aburkut** [comment]
Mainnet `Swapcicle` test results:

paraswap-dex-lib – uniswap-v2-e2e-mainnet test ts 2023-02-03 20-02-16

**Comment by aburkut** [comment]
Arbitrum chain `Swapcicle` test results:

paraswap-dex-lib – uniswap-v2-e2e-arbitrum test ts 2023-02-06 13-37-23

File: src/dex/uniswap-v2/config.ts [hunk 1]
```diff
@@ -450,12 +450,60 @@ export const UniswapV2Config: DexConfigMap<DexParams> = {
...
       initCode:
         '0x9e43ee37212e3296c7f6087d3e0a37b48a4e4e413538dac0fd18cfe2f80666c1',
       feeCode: 30,
     },
+    [Network.POLYGON]: {
+      subgraphURL:
+        'https://api.thegraph.com/subgraphs/name/swapsicledex/swapsicle-exchange-polygon',
+      factoryAddress: '0x735ab9808d792B5c8B54e31196c011c26C08b4ce',
+      initCode:
+        '0x2ed0c8714ca80192f88764ee4b4c8c8cb6dfc01859a02b25ce67f304e499d48e',
```
[the remark this answers is not in the document]
Comment (mwamedacen) [reply src/dex/uniswap-v2/config.ts:464]: @rorrieswap can advise as well

**Review by mwamedacen — APPROVED** [review]

File: src/dex/uniswap-v2/config.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mwamedacen) [reply src/dex/uniswap-v2/config.ts:464]: More importantly I diffed the pair contract code and it all matches. For some reason the creationCode differs a bit but this is minor.

[the remark this answers is not in the document]
Comment (aburkut) [reply src/dex/uniswap-v2/config.ts:464]: @mwamedacen I checked  `pairCodeHash` method  for `Swapsicle` contracts on all the listed chains.`initCode` attribute in our config contains correct values for all of the cases. 

`Avalanche`:

SicleFactory | Address 0x9C60C867cE07a3c403E2598388673C10259EC768 | SnowTrace 2023-02-07 19-02-25

`Fantom`:

SicleFactory | Address 0x98F23162E3a7FE610aC89C88E4217a599A15858F | FtmScan 2023-02-07 19-01-34


`Ethereum/polygon/bsc/optimism/arbitrum`:

SicleFactory | Address 0x2F0C7C98462651BB2102F6Cd05acDAd333E031b0 | Etherscan 2023-02-07 19-01-56

SicleFactory | Address 0x735ab9808d792B5c8B54e31196c011c26C08b4ce | PolygonScan 2023-02-07 19-00-46


SicleFactory | Address 0xEe673452BD981966d4799c865a96e0b92A8d0E45 | BscScan 2023-02-07 19-01-19

Contract Address 0x2f0c7c98462651bb2102f6cd05acdad333e031b0 | Optimism 2023-02-07 19-02-06

SicleFactory | Address 0x2f0c7c98462651bb2102f6cd05acdad333e031b0 | Arbiscan 2023-02-07 19-01-46



@mwamedacen I don't know how to check why the hash of pair's bytecode changes is not the same. Probably let's ask @rorrieswap

**Review by skypper — APPROVED** [review]

**Pull request merged**